src/v/pandaproxy/server.h

```diff
@@ -182,8 +182,14 @@ class auth_ctx_server : public ctx_server<service_t> {
 
         template<std::invocable<kafka_client_cache&> Func>
         auto dispatch(Func&& func) {
-            return service().client_cache().invoke_on_cache(
-              user, std::forward<Func>(func));
+            // Access the cache on the appropriate shard.
+            return service().client_cache().invoke_on(
+              user_shard(user.name),
+              ss::smp_submit_to_options{context().smp_sg},
+              [func{std::forward<Func>(func)}](
+                kafka_client_cache& cache) mutable {
+                  return cache.invoke_on_cache(std::forward<Func>(func));
```

Comment:
`std::move` into `invoke_on_cache`. i don't think once func is in the capture there are any value-categories preserved (although, it's going to be ok either way).